src/Components/App/App.js

```diff
@@ -7,16 +7,35 @@ import DisplayArea from '../DisplayArea/DisplayArea';
 const App = () => {
 
   const [recipes, setRecipes] = useState([]);
+  const [favorites, setFavorites] = useState([]);
 
   const addRecipes = (data) => {
     setRecipes(data)
   }
 
+  const handleFavorites = (newFavorite) => {
+    if (!favorites.includes(newFavorite)) {
+      setFavorites([...favorites, newFavorite])
+    } else {
+      const updated = favorites.filter(favorite => favorite !== newFavorite)
+      setFavorites(updated);
+    }
+  }
+
   const clearRecipes = () => {
     setRecipes('');
   }
 
   useEffect(() => {
+    if (favorites.length >= 1) {
+      localStorage.setItem('favoriteRecipes', JSON.stringify(favorites));
+    }
+  }, [favorites])
+
+  useEffect(() => {
+    if (!favorites.length) {
+      setFavorites(JSON.parse(localStorage.getItem('favoriteRecipes')))
+    }
     setRecipes('');
```

Comment:
Looks good, I know the `clearRecipes` helper function is made to simplify an `onClick` function. Maybe for consistency and cleanliness consider staying consistent with using the helper function. OR remove the helper function all together, because `setRecipe` does the exact same thing. Something to consider!

Reply:
Great point! I originally wrapped it in clearRecipes because I thought I was going to be passing it down as a prop to some of the other components. I've changed this to just call the original setRecipe on both of the links.